mono/metadata/appdomain.c

```diff
@@ -1961,7 +1961,11 @@ ves_icall_System_Reflection_Assembly_LoadFrom (MonoString *fname, MonoBoolean re
 
 		if (status == MONO_IMAGE_IMAGE_INVALID)
 			exc = mono_get_exception_bad_image_format2 (NULL, fname);
-		else
+		else if (status == MONO_IMAGE_RUNNING_REFERENCE_ASSEMBLY) {
+			char *msg = g_strdup_printf ("Could not load file or assembly %s or one of its dependencies. Reference assemblies should not be loaded for execution.  They can only be loaded in the Reflection-only loader context.", ass->aname.name);
```

Comment:
I'd change the first part to `Didn't load file or assembly %s or one of its dependencies because it is a reference assembly. ...` so it's clear that this is intentional.